packages/ui/src/working-groups/modals/AddStakeModal/AddWorkerStakeModal.tsx

```diff
@@ -0,0 +1,116 @@
+import { useMachine } from '@xstate/react'
+import BN from 'bn.js'
+import React, { useEffect, useMemo } from 'react'
+import * as Yup from 'yup'
+
+import { useBalance } from '@/accounts/hooks/useBalance'
+import { ButtonPrimary, ButtonsGroup } from '@/common/components/buttons'
+import { FailureModal } from '@/common/components/FailureModal'
+import { InputComponent, InputNumber } from '@/common/components/forms'
+import { getErrorMessage, hasError } from '@/common/components/forms/FieldError'
+import { Modal, ModalBody, ModalFooter, ModalHeader } from '@/common/components/Modal'
+import { useApi } from '@/common/hooks/useApi'
+import { useForm } from '@/common/hooks/useForm'
+import { useModal } from '@/common/hooks/useModal'
+import { useNumberInput } from '@/common/hooks/useNumberInput'
+import { formatTokenValue } from '@/common/model/formatters'
+import { AddWorkerStakeSignModal } from '@/working-groups/modals/AddStakeModal/AddWorkerStakeSignModal'
+import { addStakeMachine } from '@/working-groups/modals/AddStakeModal/machine'
+import { SuccessModal } from '@/working-groups/modals/AddStakeModal/SuccessModal'
+import { AddWorkerStakeModalCall } from '@/working-groups/modals/AddStakeModal/types'
+
+export interface IncreaseStakeFormFields {
+  amount?: string
+}
+
+const StakeFormSchema = Yup.object().shape({
+  amount: Yup.number().required(),
+})
+
+export const AddWorkerStakeModal = () => {
+  const { api } = useApi()
+  const { hideModal, modalData } = useModal<AddWorkerStakeModalCall>()
+  const [state, send] = useMachine(addStakeMachine)
+  const { minStake, stake, group, runtimeId, roleAccount } = modalData.worker
+  const minAddStake = minStake - stake
+  const [amount, setAmount] = useNumberInput(0, minAddStake)
+  const balance = useBalance(roleAccount)
+
+  const schema = useMemo(() => {
+    StakeFormSchema.fields.amount = StakeFormSchema.fields.amount.min(minAddStake, 'You need at least ${min} stake')
+    if (balance?.transferable) {
+      StakeFormSchema.fields.amount = StakeFormSchema.fields.amount.max(
+        balance?.transferable.toNumber(),
+        'Given amount exceed your transferable balance of ${max} JOY'
+      )
+    }
+    return StakeFormSchema
+  }, [minAddStake.toString(), balance?.transferable.toString()])
+
+  const { changeField, validation, fields } = useForm<IncreaseStakeFormFields>({ amount: undefined }, schema)
+  const { isValid, errors } = validation
+
+  useEffect(() => {
+    changeField('amount', amount)
+  }, [amount])
+
+  const onSubmit = () => {
+    send({ type: 'DONE', form: fields })
+  }
+
+  if (state.matches('transaction')) {
+    const transaction = api?.tx[group.id].increaseStake(runtimeId, new BN(state.context.form.amount || 0))
```

Comment:
FYI, we have: `getGroup(api, group.id)` which does basically the same. I think that for now, it would be better to stick to the `getGroup(api, group.id)`.

If you think that using `api.tx[group.id]` is better (for me both are OK now), please create a new PR that removes `getGroup()` all together.